src/OgAccess.php

```diff
@@ -310,4 +418,33 @@ public function reset() {
     $this->permissionsCache = [];
   }
 
+  /**
+   * Returns a group membership in which the user is not a member.
+   *
+   * This is intended to be used to retrieve the permissions a non-member has in
+   * the given group. Note that this membership is created on the fly and is not
+   * saved.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   *   The group to get the membership for.
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   The user to get the membership for.
+   *
+   * @return \Drupal\og\Entity\OgMembership
+   *   The OgMembership entity.
+   */
+  protected static function getNonMemberMembership(EntityInterface $group, AccountInterface $user) {
+    $role_id = implode('-', [
+      $group->getEntityTypeId(),
+      $group->bundle(),
+      OgRoleInterface::ANONYMOUS,
+    ]);
+
+    return OgMembership::create()
+      ->setUser($user)
+      ->setGroup($group)
+      ->setState(OgMembershipInterface::STATE_PENDING)
+      ->setRoles([new OgRole(['id' => $role_id])]);
```

Comment:
This is probably related to https://github.com/amitaibu/og/pull/217/files#r73741176 - For me it looks wrong. We are creating a lot of dummy data, just to get the access, which we are already doing [here](https://github.com/amitaibu/og/blob/8.x-1.x/src/OgAccess.php#L155-L160)